### PR TITLE
feat: forest-cli net info

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/3218

Changes introduced in this pull request:

- Add `forest-cli net info` command that prints libp2p stats directly from [`Swarm::network_info`](https://docs.rs/libp2p/latest/libp2p/struct.Swarm.html#method.network_info) API

Sample output:
```console
forest libp2p swarm info:
num peers: 47
num connections: 59
num pending: 3
num pending incoming: 0
num pending outgoing: 3
num established: 56
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/chain_sync/chain_muxer.rs [hunk 1]
```diff
@@ -419,6 +419,10 @@ where
                 metrics::LIBP2P_MESSAGE_TOTAL
                     .with_label_values(&[metrics::values::PEER_DISCONNECTED])
                     .inc();
+                // Unset heavist tipset for unset peers
```
Comment (LesnyRumcajs) [inline src/chain_sync/chain_muxer.rs:422]: why? what does it have to do with this subcommand?

File: src/cli/subcommands/net_cmd.rs [hunk 2]
```diff
@@ -44,6 +47,19 @@ impl NetCommands {
                 print_stdout(addresses.join("\n"));
                 Ok(())
             }
+            Self::Info => {
+                let info = net_info((), &config.client.rpc_token)
+                    .await
+                    .map_err(handle_rpc_err)?;
+                println!("forest libp2p swarm info:");
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply src/cli/subcommands/net_cmd.rs:54]: I thought about it but not sure if it's good practice to implement a multi-line `Display` message

File: src/chain_sync/chain_muxer.rs (hunk 1 above)
Comment (hanabi1224) [reply src/chain_sync/chain_muxer.rs:422]: I was using this to cross-check the correctness of the metrics, it's part of https://github.com/ChainSafe/forest/issues/3218 and I can convert it into a separate PR if you like.

Comment (LesnyRumcajs) [reply src/chain_sync/chain_muxer.rs:422]: Let's do it in a separate PR and comment on it some more, then.

Comment (hanabi1224) [reply src/chain_sync/chain_muxer.rs]: Removed

**Review by lemmih — APPROVED** [review]

**Pull request merged**